packages/ia-topnav/src/media-button.js

```diff
@@ -0,0 +1,100 @@
+import { html } from 'lit-element';
+import TrackedElement from './tracked-element';
+import icons from './assets/img/icons';
+import toSentenceCase from './lib/toSentenceCase';
+import mediaButtonCSS from './styles/media-button';
+
+class MediaButton extends TrackedElement {
+  static get styles() {
+    return mediaButtonCSS;
+  }
+
+  static get properties() {
+    return {
+      config: { type: Object },
+      icon: { type: String },
+      href: { type: String },
+      label: { type: String },
+      mediatype: { type: String },
+      openMenu: { type: String },
+      selected: { type: Boolean },
+      followable: { type: Boolean },
+    };
+  }
+
+  static get icons() {
+    return icons;
+  }
+
+  constructor() {
+    super();
+    this.config = {};
+    this.icon = '';
+    this.href = '';
+    this.label = '';
+    this.mediatype = '';
+    this.openMenu = '';
+    this.selected = false;
+    this.followable = false;
+  }
+
+  onClick(e) {
+    this.trackClick(e);
+    e.preventDefault();
+    // On desktop viewport widths, the media subnav is always visible. To
+    // ensure the media subnav is open on mobile if the viewport is
+    // resized, the openMenu needs to be set to 'media'.
+    if (this.openMenu !== 'media') {
+      this.dispatchEvent(new CustomEvent('menuToggled', {
+        bubbles: true,
+        composed: true,
+        detail: {
+          menuName: 'media'
+        }
+      }));
```

Comment:
Minor, but one thing I like to do is pull out the "dispatch" calls when they get big like this. It lets you keep the "functional" flow of this method small, ie:
```
if (this.openMenu !== 'media') {
  this.dispatchMenuToggledEvent();
}
this.dispatchMediaTypeSelectedEvent();
```